main/src/library/MutDeque.flix

```diff
@@ -442,10 +442,12 @@ mod MutDeque {
             if (i == deref back) {
                 ()
             } else {
-                match Array.nth(i, deref a) {
-                    case Some(x) => f(x)
-                    case None    => bug!("An error occurred in MutDeque.forEach!")
+                let elt = try {
+                    Array.nth(i, deref a)
+                } with IndexOutOfBounds {
+                    def throw(_, _) = bug!("An error occurred in MutDeque.forEach!")
                 };
+                f(elt);
```

Comment:
I think it woud be more natural to have 

```
try {
   f(Array.nth(i, deref a))
} catch ...
```

Reply:
My thinking was f might throw IndexOutOfBounds, and then we'd report it as a bug in forEach.

Reply:
Yes, we should still do that. Its just there is no reason for the local variable elt.

Reply:
Would using `Array.get` be ok here too?